front/temporal/agent_loop/lib/run_model.ts

```diff
@@ -508,6 +508,35 @@ export async function runModelActivity(
 
     if (shouldYieldCancel) {
       await flushParserTokens();
+
+      // Persist partial content so the UI keeps what has been generated so far.
+      const partialContent = contentParser.getContent() ?? "";
+      const partialChainOfThought =
+        (nativeChainOfThought || contentParser.getChainOfThought()) ?? "";
+
+      if (partialChainOfThought.length > 0) {
+        if (!agentMessage.chainOfThought) {
+          agentMessage.chainOfThought = "";
+        }
+        agentMessage.chainOfThought += partialChainOfThought;
+      }
+      if (partialContent.length) {
+        agentMessage.content = (agentMessage.content ?? "") + partialContent;
+
+        // Persist partial text content so reload retains it
+        await AgentStepContentResource.createNewVersion({
```

Comment:
Are we sure we want to do that here?
I'm not familiar with this new code but looking quickly wouldn't it be more consistent to do it in `processEventForDatabase` that is called from `updateResourceAndPublishEvent` that we call just below?